### PR TITLE
[libdeflate] Add new port

### PR DESCRIPTION
**Pull request by SeekingMeaning**
- Partially fixes issue #9424
- Currently only supports Windows

### CODE REVIEW

File: ports/libdeflate/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,65 @@
+include(vcpkg_common_functions)
```
Comment (NancyLi1013) [inline ports/libdeflate/portfile.cmake]: `include(vcpkg_common_functions)` is no longer needed.
Could you please remove it?

File: ports/libdeflate/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+)
+message(STATUS "Build ${TARGET_TRIPLET}-dbg done")
+
+if (VCPKG_LIBRARY_LINKAGE STREQUAL "dynamic")
+    file (COPY ${SOURCE_PATH}/libdeflate.dll DESTINATION ${CURRENT_PACKAGES_DIR}/debug/bin)
+    file (COPY ${SOURCE_PATH}/libdeflate.lib DESTINATION ${CURRENT_PACKAGES_DIR}/debug/lib)
+elseif (VCPKG_LIBRARY_LINKAGE STREQUAL "static")
+    file (INSTALL ${SOURCE_PATH}/libdeflatestatic.lib DESTINATION ${CURRENT_PACKAGES_DIR}/debug/lib RENAME libdeflate.lib)
+endif()
+
```
Comment (NancyLi1013) [inline ports/libdeflate/portfile.cmake]: Could you use `vcpkg_configure_qmake()` and `vcpkg_install_qmake()` to refactor these code?
Please refer to [libqglviewer](https://github.com/Microsoft/vcpkg/blob/master/ports/libqglviewer/portfile.cmake),
[qcustomplot](https://github.com/microsoft/vcpkg/blob/master/ports/qcustomplot/portfile.cmake).

File: ports/libdeflate/portfile.cmake [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+file(
+    COPY ${SOURCE_PATH}/libdeflate.h
+    DESTINATION ${CURRENT_PACKAGES_DIR}/include
+)
+file(INSTALL ${SOURCE_PATH}/COPYING DESTINATION ${CURRENT_PACKAGES_DIR}/share/libdeflate RENAME copyright)
```
Comment (NancyLi1013) [inline ports/libdeflate/portfile.cmake]: Could you update above line as `file(INSTALL ${SOURCE_PATH}/COPYING DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)`?

File: ports/libdeflate/portfile.cmake [hunk 4]
```diff
@@ -0,0 +1,65 @@
+include(vcpkg_common_functions)
+
+if(NOT VCPKG_TARGET_IS_WINDOWS)
+  message(FATAL_ERROR "This port is only for Windows Desktop")
+endif()
```
Comment (NancyLi1013) [inline ports/libdeflate/portfile.cmake]: If this port only supports Windows, Please use `vcpkg_fail_port_install(ON_TARGET "Linux" "OSX")` instead.

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

**Comment by NancyLi1013** [comment]
Hi @SeekingMeaning thanks for thsi PR.
I noticed that this port failed on` x64-uwp` and `arm-uwp` triplets from CI test result.
Are those expected? If yes, please add the failure to `vcpkg_fail_port_install(ON_TARGET "UWP")`
If not, please try to fix them.

Besides this, Please update [baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) with the following contents:
` libdefalte:x64-linux=fail`
`libdefalte:x64-osx=fail`

Note: If `UWP `is not supported, please also add `libdefalte:x64-uwp=fail `and `libdeflate:arm-uwp=fail` to the fail list in above baseline file.

**Comment by SeekingMeaning** [comment]
Anything else? I'm not sure why OS X failed since Linux passes now.

**Comment by SeekingMeaning** [comment]
Closing this since it has been combined with #9569

**Pull request closed without merging**